tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/configurables/ConfigurableTest.java

```diff
@@ -127,15 +127,43 @@ public void testFileEnvVariableBasedConfigurable() throws BallerinaTestException
+        LogLeecher errorLeecher3 = new LogLeecher("[BAL_CONFIG_DATA:(1:22,1:24)] missing new line", ERROR);
+        LogLeecher errorLeecher4 = new LogLeecher("[BAL_CONFIG_DATA:(1:36,1:39)] missing new line", ERROR);
+        LogLeecher errorLeecher5 = new LogLeecher("[BAL_CONFIG_DATA:(1:52,1:57)] missing new line", ERROR);
+        LogLeecher errorLeecher6 = new LogLeecher("[BAL_CONFIG_DATA:(1:71,1:75)] missing new line", ERROR);
+        LogLeecher errorLeecher7 = new LogLeecher("[BAL_CONFIG_DATA:(1:89,1:94)] missing new line", ERROR);
+        LogLeecher errorLeecher8 = new LogLeecher("[BAL_CONFIG_DATA:(1:104,1:111)] missing new line", ERROR);
+        LogLeecher errorLeecher9 = new LogLeecher("error: value not provided for required configurable variable" +
+                " 'stringVar'", ERROR);
+        LogLeecher errorLeecher10 = new LogLeecher("error: value not provided for required configurable variable" +
+                " 'booleanVar'", ERROR);
+        LogLeecher errorLeecher11 = new LogLeecher("error: value not provided for required configurable variable" +
+                " 'intArr'", ERROR);
+        LogLeecher errorLeecher12 = new LogLeecher("error: value not provided for required configurable variable" +
+                " 'decimalArr'", ERROR);
         // test configuration through `BAL_CONFIG_DATA` env variable
         String configData = "[envVarPkg] intVar = 42 floatVar = 3.5 stringVar = \"abc\" booleanVar = true " +
-                "decimalVar = 24.87 intArr = [1,2,3] floatArr = [9.0, 5.6] " +
-                "stringArr = [\"red\", \"yellow\", \"green\"] booleanArr = [true, false,false, true] " +
-                "decimalArr = [8.9, 4.5, 6.2]";
-        executeBalCommand("", "envVarPkg",
-                addEnvironmentVariables(Map.ofEntries(Map.entry(CONFIG_DATA_ENV_VARIABLE, configData))));
-
+                "decimalVar = 24.87 intArr = [1,2,3] floatArr = [9.0, 5.6]";
+        bMainInstance.runMain(testFileLocation, "envVarPkg", null, new String[]{},
+                addEnvironmentVariables(Map.ofEntries(Map.entry(CONFIG_DATA_ENV_VARIABLE, configData))), null,
+                new LogLeecher[]{errorLeecher1, errorLeecher2, errorLeecher3, errorLeecher4, errorLeecher5,
+                        errorLeecher6, errorLeecher7, errorLeecher8, errorLeecher9, errorLeecher10,
+                        errorLeecher11, errorLeecher12});
+        errorLeecher1.waitForText(5000);
+        errorLeecher2.waitForText(5000);
+        errorLeecher3.waitForText(5000);
+        errorLeecher4.waitForText(5000);
+        errorLeecher5.waitForText(5000);
+        errorLeecher6.waitForText(5000);
+        errorLeecher7.waitForText(5000);
+        errorLeecher8.waitForText(5000);
+        errorLeecher9.waitForText(5000);
+        errorLeecher10.waitForText(5000);
+        errorLeecher11.waitForText(5000);
+        errorLeecher12.waitForText(5000);
```

Comment:
Do we need 5s here for each? Can we reduce and see?

Reply:
sure.
Even though, it says to wait for 5s, it will execute further once the log is received.

Reply:
I think then we can keep it like this because this will not get delayed due to the `5000` value, and we will have less intermittent failures due to high timeouts.

Reply:
+1